crates/proto/src/error.rs

```diff
@@ -211,6 +211,10 @@ pub enum ProtoErrorKind {
     #[error("algorithm type value unknown: {0}")]
     UnknownAlgorithmTypeValue(u8),
 
+    /// An unknown digest type was found
+    #[error("digest type value unknown: {0}")]
+    UnknownDigestTypeValue(u8),
```

Comment:
Can/should we stuff this in `DnsSecError` instead, or does that cause problems?

Reply:
I'm effectively splitting this variant out from `UnknownAlgorithmTypeValue` above, and thus making each variant more particular about which category of algorithm it refers to.

It looks like `DnsSecError` only gets encapsulated in `hickory_client::Error` and `hickory_proto::dnssec::proof::ProofError`, so that wouldn't work with the current signature of `Nsec3HashAlgorithm::hash()`, which may return a `ProtoError`.